File: SFC CODE/SFC CODE/SysBasicClass/Control/InputText.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevComponents.DotNetBar;
using DevComponents.WinForms;
using DevComponents.DotNetBar.Controls;
namespace SysBasicClass
{
    public partial class InputText : Form
    {
        MaskedTextBoxAdv _text;
        public InputText()
        {
            InitializeComponent();
        }
        public InputText(MaskedTextBoxAdv ObjText)
        {
            InitializeComponent();
            _text = ObjText;
            StringBuilder str = new StringBuilder();
            if (ObjText.Text.ToString().Trim() != "")
            {
                string[] strarry = ObjText.Text.ToString().Trim().Split(';');
                for (int i = 0; i < strarry.Length; i++)
                {
                    str.Append(strarry[i].ToString().Trim());
                    str.AppendLine();
                }
            }
            textBoxX1.Text = str.ToString();            
        }

        private void button1_Click(object sender, EventArgs e)
        {
       
            string txt = "";
            if (textBoxX1.Text != "")
            {
                int i = 0;
                for (i = 0; i < textBoxX1.Lines.Length; i++)
                {
                    if (textBoxX1.Lines[i].ToString().Trim() != "")
                    {
                        txt = txt + textBoxX1.Lines[i].ToString().Trim() + ";";
                    }
                }
                txt = txt.Substring(0, txt.Length - 1);
            }
            _text.Text = txt;
            this.Close();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: SFC CODE/SFC CODE/BasicFind/BasicFind/InputText.vb
Imports DevComponents.DotNetBar
Imports DevComponents.WinForms
Imports DevComponents.DotNetBar.Controls
Imports System.Text

Public Class InputText
    Dim _text As MaskedTextBoxAdv
    Public Sub New(ByVal ObjText As MaskedTextBoxAdv)

        ' 此调用是设计器所必需的。
        InitializeComponent()

        ' 在 InitializeComponent() 调用之后添加任何初始化。
        _text = ObjText
        Dim str As New StringBuilder()
        If ObjText.Text.ToString().Trim() <> "" Then
            Dim strarry As String() = ObjText.Text.ToString().Trim().Split(";")
            For i = 0 To strarry.Length - 1
                str.Append(strarry(i).ToString().Trim())
                str.AppendLine()
            Next
        End If
        textBoxX1.Text = str.ToString()
    End Sub
    
    Private Sub button1_Click(sender As Object, e As EventArgs) Handles button1.Click
        Dim txt As String = ""
        If (textBoxX1.Text <> "") Then
            For i = 0 To textBoxX1.Lines.Length - 1
                If textBoxX1.Lines(i).ToString().Trim() <> "" Then
                    txt = txt + textBoxX1.Lines(i).ToString().Trim() + ";"
                End If
            Next
            txt = txt.Substring(0, txt.Length - 1)
        End If
        _text.Text = txt
        Me.Close()
    End Sub

    Private Sub button2_Click(sender As Object, e As EventArgs) Handles button2.Click
        Me.Close()
    End Sub
End Class